test/es6/ProxyPropertiesAPI.js

```diff
@@ -0,0 +1,90 @@
+            let properties = WScript.GetProxyProperties(undefined);
+            assert.isUndefined(properties, "ProxyProperties of undefined should be undefined");
+            properties = WScript.GetProxyProperties(1);
+            assert.isUndefined(properties, "ProxyProperties of number should be undefined");
+            properties = WScript.GetProxyProperties({});
+            assert.isUndefined(properties, "ProxyProperties of non-proxy object should be undefined");
+        }
+    },
+    {
+        name: "GetProxyProperties: revocable Proxy",
+        body: function () {
+            let revocable = Proxy.revocable({someProperty: true, otherProperty: false}, {otherProperty: true, newProperty: 5});
+            let proxy = revocable.proxy;
+            let properties = WScript.GetProxyProperties(proxy);
+
+            let names = Object.getOwnPropertyNames(properties);
+            assert.areEqual(names.length, 3, "proxy properties names should have length 3.");
+            assert.isTrue(names.includes("target"));
+            assert.isTrue(names.includes("handler"));
+            assert.isTrue(names.includes("revoked"));
+            assert.isFalse(properties.revoked, "Revoked bool should be false.");
+
+            names = Object.getOwnPropertyNames(properties.target);
+            assert.areEqual(names.length, 2, "proxy properties target names should have length 2.");
+            assert.areEqual(properties.target.someProperty, true);
+            assert.areEqual(properties.target.otherProperty, false);
+            
+            names = Object.getOwnPropertyNames(properties.handler);
+            assert.areEqual(names.length, 2, "proxy properties handler names should have length 2.");
+            assert.areEqual(properties.handler.newProperty, 5);
+            assert.areEqual(properties.handler.otherProperty, true);
+
+            revocable.revoke();
+            properties = WScript.GetProxyProperties(proxy);
+
+            names = Object.getOwnPropertyNames(properties);
+            assert.areEqual(names.length, 3, "proxy properties names for revokes proxy should have length 2.");
+            assert.isTrue(names.includes("target"));
+            assert.isTrue(names.includes("handler"));
+            assert.isTrue(properties.revoked, "Revoked bool should be true.");
+
+            assert.isUndefined(properties.target, "Target of revoked proxy should be undefined.");
+            assert.isUndefined(properties.handler, "Handler of revoked proxy should be undefined.");
+        }
+    },
+    {
+        name: "GetProxyProperties: normal Proxy",
+        body: function () {
+            let proxy = new Proxy({someProperty: true, otherProperty: false}, {otherProperty: true, newProperty: 5});
+            let properties = WScript.GetProxyProperties(proxy);
+
+            let names = Object.getOwnPropertyNames(properties);
+            assert.areEqual(names.length, 3, "proxy properties names should have length 2");
```

Comment:
>length 2 [](start = 81, length = 8)

shoul have 3 ?

Reply:
should*

---
In reply to: [174286438](https://github.com/Microsoft/ChakraCore/pull/4806#discussion_r174286438) [](ancestors = 174286438)